WordPress/gutenberg
# Asynchronously load tinyMCE the first time a classic block is edited

## Description

Added `LazyLoad` component. This component allows us to wrap existing components and delay them until required 3rd party scripts and styles are asynchronously loaded. It also provides a hook for post-load setup that resolves before the children are rendered, guaranteeing that the children aren't active until the world around them is set up correctly.

I've then wrapped `ClassicEdit` in `LazyLoad`. This 100% does not work and is just a proof of concept.

A related PR: https://github.com/WordPress/gutenberg/pull/21244

We'd replace `load-scripts/styles` in the current code here with requests to those endpoints.

## Types of changes
New feature

## Checklist:

## Testing instructions

1. Checkout https://github.com/WordPress/wordpress-develop/pull/232 locally and configure Gutenberg to point towards `wordpress-develop` by following the instructions for `wp-env` configuration here: https://github.com/WordPress/gutenberg/blob/master/packages/env/README.md#wp-envjson. Run WordPress using `npx wp-env start` inside your Gutenberg repo.
2. Checkout this PR locally and run `npm run dev`
3. Open the block editor and search the page source for the TinyMCE source code; ensure it does not exist. Additionally, check that `window.tinymce === undefined`
4. Edit a classic block and watch the network tab in your browser. Note that TinyMCE is loaded and `window.tinymce` is defined.
5. Ensure TinyMCE translations are loading correctly by changing your user's locale to one with different translations than what you're starting with and confirming that the translations worked.
6. Confirm that exiting a classic block and re-entering it (or another classic block) does _not_ make a new request for TinyMCE or the translations.
